internal/provider/resource_host_set_plugin.go

```diff
@@ -131,6 +132,25 @@ func setFromHostSetPluginResponseMap(d *schema.ResourceData, raw map[string]inte
 		attrRaw, ok := raw["attributes"]
 		switch ok {
 		case true:
+			if attrMap, ok := attrRaw.(map[string]interface{}); ok {
+				for attrKey := range attrMap {
+					// Flatten attribute filters data structure from Boundary SDK to match terraform input data structure
+					// This sets the value which is used for `attributes_json` diff checker
+					if attrKey == "filters" {
+						if filters, ok := attrMap[attrKey].([]interface{}); ok {
+							var flattenedFilters []string
+							for _, f := range filters {
+								if filter, ok := f.(string); ok {
+									flattenedFilters = append(flattenedFilters, filter)
+								}
+							}
+							attrMap[attrKey] = strings.Join(flattenedFilters, ",")
+						}
+					}
+				}
```

Comment:
I decided to iterate over it in case there are other attributes/fields in the future that will need to be changed to match what Boundary TF provider expects. There aren't a lot of attributes so I did not think iterating over it will be a big issue. To keep it simple, we can just get filters directly as suggested as well.

Reply:
Could we not just drop `filters` completely and flatten any attribute that is a `if toFlatten, ok := filtersInt.([]interface{})`

Reply:
Yes we could definitely do that but I am not sure if we definitely want to flatten all attributes. Just being safer by trying to flatten just filters for now.